tests/test_aaa_authorization_service.rb

```diff
@@ -22,16 +22,17 @@ class TestAaaAuthorizationService < CiscoTestCase
 
   def setup
     super
-    cleanup_aaa if @@pre_clean_needed
-    @@pre_clean_needed = false # rubocop:disable Style/ClassVars
-    feature_tacacs
+    if @@pre_clean_needed
+      cleanup_aaa
+      config('feature tacacs+')
+      @@pre_clean_needed = false # rubocop:disable Style/ClassVars
+    end
     preconfig_tacacs_server_access(tacacs_groups[0])
     config_tacacs_servers(tacacs_groups[1..3])
   end
 
   def teardown
     cleanup_aaa
-    feature_tacacs(false)
```

Comment:
Our stance is for the provider object itself but IMO, does not apply to test cases.  Any configuration needed by a test or set of tests should be removed at the end.  While there *may* not be any perceptible impact now, this could certainly change in the future as if/when changes are made to the tacacs feature.  If you are worried about time, I came across the `after_tests()` method that might allow us to just cleanup at the end after all of the tests have completed.

- https://github.com/seattlerb/minitest/search?utf8=%E2%9C%93&q=after_tests